src/precice/tests/SerialTests.cpp

```diff
@@ -1400,6 +1399,171 @@ BOOST_AUTO_TEST_CASE(testPreconditionerBug,
+
+  if (utils::Parallel::getProcessRank() == 0) {
+    // Expected values in the target solver
+    double expectedValueA = 3.0;
+    double expectedValueB = 7.0;
+    double expectedValueC = 11.0;
+    double expectedValueD = 15.0;
+    
+    // Target solver
+    SolverInterface cplInterface("SolverTarget", configFile, 0, 1);
+
+    // Set mesh
+    Vector3d coordA{0.0, 0.0, 0.3};
+    Vector3d coordB{1.0, 0.0, 0.3};
+    Vector3d coordC{1.0, 1.0, 0.3};
+    Vector3d coordD{0.0, 1.0, 0.3};
+
+    const int meshID = cplInterface.getMeshID("MeshTarget");
+
+    int idA = cplInterface.setMeshVertex(meshID, coordA.data());
+    int idB = cplInterface.setMeshVertex(meshID, coordB.data());
+    int idC = cplInterface.setMeshVertex(meshID, coordC.data());
+    int idD = cplInterface.setMeshVertex(meshID, coordD.data());
+
+    int idAB = cplInterface.setMeshEdge(meshID, idA, idB);
+    int idBC = cplInterface.setMeshEdge(meshID, idB, idC);
+    int idCD = cplInterface.setMeshEdge(meshID, idC, idD);
+    int idDA = cplInterface.setMeshEdge(meshID, idD, idA);
+    int idCA = cplInterface.setMeshEdge(meshID, idC, idA);
+
+    cplInterface.setMeshTriangle(meshID, idAB, idBC, idCA);
+    cplInterface.setMeshTriangle(meshID, idCD, idDA, idCA);
+
+    // Initialize, the mesh
+    double dt = cplInterface.initialize();
+
+    // Read the summed data from the mesh.
+    int    dataAID = cplInterface.getDataID("Target", meshID);
+    double valueA, valueB, valueC, valueD;
+
+    while(cplInterface.isCouplingOngoing()){ 
+      dt = cplInterface.advance(dt);
+      
+      cplInterface.readScalarData(dataAID, idA, valueA);
+      cplInterface.readScalarData(dataAID, idB, valueB);
+      cplInterface.readScalarData(dataAID, idC, valueC);
+      cplInterface.readScalarData(dataAID, idD, valueD);
+    }
+
+    BOOST_TEST(valueA == expectedValueA);
+    BOOST_TEST(valueB == expectedValueB);
+    BOOST_TEST(valueC == expectedValueC);
+    BOOST_TEST(valueD == expectedValueD);
+
+    cplInterface.finalize();
+  }
+  else if (utils::Parallel::getProcessRank() == 1) {
+    // Source solver one
+    SolverInterface cplInterface("SolverSourceOne", configFile, 0, 1);
+
+    // Set mesh
+    Vector3d coordA{0.0, 0.0, 0.3};
+    Vector3d coordB{1.0, 0.0, 0.3};
+    Vector3d coordC{1.0, 1.0, 0.3};
+    Vector3d coordD{0.0, 1.0, 0.3};
+
+    const int meshID = cplInterface.getMeshID("MeshOne");
+
+    int idA = cplInterface.setMeshVertex(meshID, coordA.data());
+    int idB = cplInterface.setMeshVertex(meshID, coordB.data());
+    int idC = cplInterface.setMeshVertex(meshID, coordC.data());
+    int idD = cplInterface.setMeshVertex(meshID, coordD.data());
+
+    int idAB = cplInterface.setMeshEdge(meshID, idA, idB);
+    int idBC = cplInterface.setMeshEdge(meshID, idB, idC);
+    int idCD = cplInterface.setMeshEdge(meshID, idC, idD);
+    int idDA = cplInterface.setMeshEdge(meshID, idD, idA);
+    int idCA = cplInterface.setMeshEdge(meshID, idC, idA);
```

Comment:
No need to define edges for this testcase. We only need edges for a nearest-projection mapping. For this test we can just use matching meshes together with a nearest-neighbor mapping.